src/modm/platform/uart/sam/uart_hal_impl.hpp.in

```diff
@@ -32,29 +35,29 @@ modm::platform::{{ name }}::setParity(const Parity parity)
 }
 
 void
-modm::platform::{{ name }}::enable()
+{{ name }}::enable()
 {
 	{{ peripheral }}->USART.CTRLA.bit.ENABLE = true;	// Uart Enable
 	while({{ peripheral }}->USART.SYNCBUSY.bit.ENABLE); // Wait for sync.
 }
 
 void
-modm::platform::{{ name }}::disable()
+{{ name }}::disable()
 {
 	{{ peripheral }}->USART.CTRLA.bit.ENABLE = false;	// Uart Disable
 	while({{ peripheral }}->USART.SYNCBUSY.bit.ENABLE); // Wait for sync.
 }
 
 void
-modm::platform::{{ name }}::reset()
+{{ name }}::reset()
 {
 	{{ peripheral }}->USART.CTRLA.bit.SWRST = true;	// Uart Reset
 	while({{ peripheral }}->USART.SYNCBUSY.bit.SWRST); // Wait for sync.
 }
 
-template<class SystemClock, modm::baudrate_t baudrate>
-void modm_always_inline
-modm::platform::{{ name }}::initialize(Parity parity)
+template<class SystemClock, modm::baudrate_t baudrate, modm::percent_t tolerance>
```

Comment:
If `tolerance` is added here, shouldn't we actually check it? Adding the baudrate calculation with compile-time checking would be quite straight forward, it's pretty similar to STM32.

There are also 8x and 16x oversampling modes, selected by setting `{{ peripheral }}->USART.CTRLA.bit.SAMPR` (16x: `0x1`, 8x: `0x3`), [see SAMD5/E5 manual, page 830,854,862](https://ww1.microchip.com/downloads/en/DeviceDoc/SAM_D5x_E5x_Family_Data_Sheet_DS60001507G.pdf). The prescaler has a 3 bit fractional and 13 bit integral part for both x8 and x16 oversampling (configured in line 87, 88 of this file).

It looks like there is  a copy-paste bug in these lines anyway, they always configure the instance `SERCOM0->USART`, not `{{ peripheral }}->USART`. That should definitively be fixed.

Reply:
I fixed this, but I still need to test this in hardware, since it may be off by a factor or 2.